tests/automl/test_automl_image_object_detection_model_wrapper.py

```diff
@@ -0,0 +1,103 @@
+# ---------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# ---------------------------------------------------------
+
+"""Tests for wrap_model function on vision-based models"""
+
+import copy
+import json
+import os
+import sys
+import tempfile
+
+import azureml.automl.core.shared.constants as shared_constants
+import mlflow
+import pytest
+import torch
+from azureml.automl.dnn.vision.object_detection.common.constants import \
+    ModelNames
+from azureml.automl.dnn.vision.object_detection.models import object_detection_model_wrappers
+from azureml.automl.dnn.vision.common.mlflow.mlflow_model_wrapper import \
+    MLFlowImagesModelWrapper
+from azureml.automl.dnn.vision.common.model_export_utils import (
+    _get_mlflow_signature, _get_scoring_method)
+from common_vision_utils import load_base64_images, load_object_fridge_dataset
+from ml_wrappers import wrap_model
+from ml_wrappers.common.constants import ModelTask
+from wrapper_validator import validate_wrapped_object_detection_model
+
+
+@pytest.mark.usefixtures('_clean_dir')
+class TestImageModelWrapper(object):
+    # Skip for older versions of python as azureml-automl-dnn-vision works with ">=3.7,<3.9"
+    @pytest.mark.skipif(sys.version_info < (3, 7),
+                        reason='azureml-automl-dnn-vision not supported for older versions')
+    @pytest.mark.skipif(sys.version_info >= (3, 9),
+                        reason='azureml-automl-dnn-vision not supported for newer versions')
+    def test_wrap_automl_object_detection_model(self):
+        data = load_object_fridge_dataset()[1:4]
+        model_name = ModelNames.FASTER_RCNN_RESNET50_FPN
+
+        with tempfile.TemporaryDirectory() as tmp_output_dir:
+
+            task_type = shared_constants.Tasks.IMAGE_OBJECT_DETECTION
+            number_of_classes = 4
+            class_names = ['can', 'carton', 'milk_bottle', 'water_bottle']
+            model_wrapper = object_detection_model_wrappers.ObjectDetectionModelFactory().get_model_wrapper(number_of_classes, model_name)
```

Comment:
Curious if this long a file is allowed under flake8. Can check and fix as needed.